### ISSUE TITLE
Docs: correct example of parameter-properties

### ISSUE THREAD

**Issue by miZyind** [typescript-eslint/typescript-eslint#4878]
### Suggested Changes

I got this error when I first try this new rule by following the example on the document:
```
Configuration for rule "@typescript-eslint/parameter-properties" is invalid:
Value "parameter-properties" should be equal to one of the allowed values.
```
and here is my config:
```
"@typescript-eslint/parameter-properties": [
  "error",
  {
    "prefer": "parameter-properties"
  }
]
```
I checked the [source code](https://github.com/typescript-eslint/typescript-eslint/pull/4622/files#diff-a0997bb21a068d001530cb7d430c5c659e8ba9ac5d60c9b44eb4a11e98003572R59).
And found that the correct values of the `prefer` option should be: `class-property`, `parameter-property`.
However, the document shows `class-properties` and `parameter-properties`.

According to the schema of the [source code](https://github.com/typescript-eslint/typescript-eslint/pull/4622/files#diff-a0997bb21a068d001530cb7d430c5c659e8ba9ac5d60c9b44eb4a11e98003572R68), the `allows` option should be corrected to `allow` as well.

I've prepared for the PR: https://github.com/miZyind/typescript-eslint/tree/docs/parameter-properties

### Affected URL(s)

https://typescript-eslint.io/rules/parameter-properties

**Comment by dora38** [comment]
@miZyind Great work! you save my confusion. thanks.

I think the following line would also be corrected. no plural, no array.
```yaml
Examples of code for the `{ "prefer": ["parameter-properties"] }` option:
```

**Comment by miZyind** [comment]
Oops,  looks like I missed that part! Thanks for your help @dora38, PR is ready.

**Comment by dora38** [comment]
@miZyind Thank you for your quick response.
two comments to #4927...

```yaml
  "@typescript-eslint/parameter-property": [
```

This might be overmuch because the rule name is plural. 

```yaml
Examples of code for the `{ "prefer": ["parameter-property"] }` option:
```

Unlike `"allow"`, `"prefer"` does not take an array.
This should be `{ "prefer": "parameter-property" }`.

**Comment by miZyind** [comment]
I just reverted the change of the rule name, and fixed the invalid example of `prefer`.
Hope the document is all correct now!
Thanks for your feedback again!

**Comment by miZyind** [comment]
I just reverted the change of the rule name, and fixed the invalid example of `prefer`.
Hope the document is all correct now!
Thanks for your feedback again! @dora38
